### PR TITLE
distutils-r1.eclass: next batch of changes

### PR DESCRIPTION
**Pull request by mgorny**
CC @gentoo/python 

Mostly meant as a place to collect minor changes.

### CODE REVIEW

File: eclass/distutils-r1.eclass [hunk 1]
```diff
@@ -425,6 +401,9 @@ unset -f _distutils_set_globals
 # on the sources directly, prepending setup.py arguments with
 # 'build --build-base ${BUILD_DIR}' to enforce keeping & using built
 # files in the specific root.
+#
+# In-source builds are deprecated and no longer supported in PEP517
+# mode.
```
Comment (ulm) [inline eclass/distutils-r1.eclass:406]: Very minor point, it is "PEP 517 mode" (with a space) in the documentation of `DISTUTILS_USE_PEP517` above. Maybe this should be consistent within the eclass?

**Review by ulm — APPROVED** [review]

**Comment by mgorny** [comment]
Ok, the `python-r1` change is actually something useful, so let's put this for review.

**Comment by mgorny** [comment]
(the CI failures are just new deprecation warnings)

**Review by arthurzam — APPROVED** [review]

File: eclass/distutils-r1.eclass [hunk 2]
```diff
@@ -733,124 +703,8 @@ esetup.py() {
...
-# If your package does not work with the other two modes but works with
-# this one, please report a bug.
-#
-# Please note that in order to test the solution properly you need
-# to unmerge the package first.
-#
-# This function is not available in PEP517 mode.  The eclass provides
-# a venv-style install unconditionally and therefore it should no longer
-# be necessary.
 distutils_install_for_testing() {
```
Comment (thesamesam) [inline eclass/distutils-r1.eclass:706]: Can you mark it as deprecated (https://devmanual.gentoo.org/eclass-writing/index.html#eclass-functions)?

**Review by thesamesam — APPROVED** [review]

File: eclass/distutils-r1.eclass (hunk 2 above)
Comment (mgorny) [reply eclass/distutils-r1.eclass:711]: Done. Note that `eclass-to-manpage.awk` doesn't really implement it, so it just shows as a regular function with the description saying it's dead.

Comment (thesamesam) [reply eclass/distutils-r1.eclass:711]: np, thanks

**Pull request closed without merging**